app/controllers/users_controller.rb

```diff
@@ -0,0 +1,50 @@
+class UsersController < ApplicationController
+  def login_form
+    @user = User.new
+  end
+
+  def login
+    name = params[:user][:name]
+    user = User.find_by(name: name)
+    if user.nil?
+      user = User.create(name: name)
+      session[:user_id] = user.id
+      flash[:success] = "Successfully created new user #{user.name} with ID #{user.id}"
+    else
+      session[:user_id] = user.id
+      flash[:success] = "Successfully logged in as existing user #{user.name}"
+    end
+
+    if user.id
```

Comment:
With our understanding of Rails, this `if ... else` checks on essentially `user.nil?`, which is covered above. It probably might feel more useful to bring this check on failure of saving inside of the first `if` after checking if `user = User.create(name: name)` succeeds or fails.